src/odin/narrativebuilder.cc

```diff
@@ -978,14 +1047,30 @@ std::string NarrativeBuilder::FormTurnInstruction(Maneuver& maneuver) {
 
   // Determine which phrase to use
   uint8_t phrase_id = 0;
-  if (!street_names.empty()) {
-    phrase_id = 1;
-  }
-  if (!begin_street_names.empty()) {
-    phrase_id = 2;
-  }
-  if (maneuver.to_stay_on()) {
-    phrase_id = 3;
+  std::string junction_name;
+  std::string guide_sign;
+
+  if (maneuver.HasGuideSign()) {
+    // Set the toward phrase - it takes priority over street names and junction name
+    phrase_id = 5;
+    // Assign guide sign
+    guide_sign = maneuver.signs().GetGuideString(element_max_count, limit_by_consecutive_count);
+  } else if (maneuver.HasJunctionNameSign()) {
+    // Set the junction phrase - it takes priority over street names
+    phrase_id = 4;
+    // Assign guide sign
+    junction_name =
+        maneuver.signs().GetJunctionNameString(element_max_count, limit_by_consecutive_count);
+  } else {
+    if (!street_names.empty()) {
```

Comment:
Since only one of these will affect the value set in `phrase_id`, an `if-else` block would read better.

Reply:
all three need to be in the `else` block

Reply:
Yeah, I was talking about the 3 `if`s in the `else` block. It looks like we only set `phrase_id` if one of the conditions is true:
```c++
if (maneuver.to_stay_on()) {
   phrase_id = 3;
} else if (!begin_street_names.empty()) {
   phrase_id = 2;
} else if (!street_names.empty()) {
   phrase_id = 1;
}
```

Reply:
I cleaned up - thanks @mandeepsandhu